File: frontend/src/services/idcard.service.js
import { idCardApi } from '../api/idcard.api';

export const idCardService = {
  // Returns null if the employee doesn't have a card issued yet, instead of throwing —
  // callers can use this to decide whether to show an "Issue Card" action.
  async getByEmployee(employeeId) {
    try {
      const res = await idCardApi.getByEmployee(employeeId);
      return res.data;
    } catch {
      return null;
    }
  },

  async create(employeeId, payload) {
    const res = await idCardApi.create(employeeId, payload);
    return res.data;
  },

  async list({ page = 0, size = 8, status } = {}) {
    const res = await idCardApi.getAll({ page, size, status: status || undefined });
    return res.data; // PageResponse<IdCardResponse>
  },

  async update(id, payload) {
    const res = await idCardApi.update(id, payload);
    return res.data;
  },

  async remove(id) {
    await idCardApi.remove(id);
  },
};

File: src/main/java/com/hrms/hr_payroll_management_system/service/IdCardService.java
package com.hrms.hr_payroll_management_system.service;

import com.hrms.hr_payroll_management_system.common.pagination.PageResponse;
import com.hrms.hr_payroll_management_system.dto.request.idcard.CreateIdCardRequest;
import com.hrms.hr_payroll_management_system.dto.request.idcard.UpdateIdCardRequest;
import com.hrms.hr_payroll_management_system.dto.response.idcard.IdCardResponse;
import com.hrms.hr_payroll_management_system.enums.IdCardStatus;

public interface IdCardService {

    IdCardResponse create(Long employeeId, CreateIdCardRequest request);

    IdCardResponse getById(Long id);

    IdCardResponse getByEmployee(Long employeeId);

    PageResponse<IdCardResponse> getAll(int page, int size, IdCardStatus status);

    IdCardResponse update(Long id, UpdateIdCardRequest request);

    void delete(Long id);
}